docs/designs/transaction-design.md

```diff
@@ -0,0 +1,218 @@
+```
+
+**Stored Procedures:**
+- **RecordTransaction**:  
+  - Inserts a new transaction record into the `Transactions` table.
+  - Parameters: `user_id`, `transaction_id`, `item_id`, `transaction_type`, `quantity`.
+  
+```sql
+CREATE PROCEDURE LogTransaction
+    @user_id INT,                     
+    @transaction_id UNIQUEIDENTIFIER, 
+    @item_id INT,                   
+    @transaction_type VARCHAR(50),     
+    @quantity INT       
+AS
+BEGIN
+    -- Insert a new record into the Transactions table
+    INSERT INTO Transactions (
+        user_id,
+        transaction_id,
+        item_id,
+        transaction_type,
+        quantity,
+    )
+    VALUES (
+        @user_id,                     
+        @transaction_id,           
+        @item_id,                   
+        @transaction_type,             
+        @quantity,                             
+    );
+END;
+```
+
+- **GetTransactionHistory**:
+- Retrieves all transactions associated with a specific `user_id`(admin user only).
+
+```sql
+CREATE PROCEDURE GetTransactionHistory
+    @AdminUserID INT                 -- The ID of the admin requesting transaction history
+AS
+BEGIN
+    -- Check if the requesting user is an admin
+    IF EXISTS (SELECT 1 FROM Users WHERE user_id = @AdminUserID AND IsAdmin = 1)
+    BEGIN
+        -- Retrieve all transaction history records
+        SELECT 
+            th.TransactionID,
+            th.user_id,
+            th.TransactionGUID,
+            th.item_id,
+            th.transaction_type,
+            th.TotalAmount,
+            th.transaction_date
+        FROM 
+            Transactions th
+        INNER JOIN 
+            Items i ON th.item_id = i.item_id
+        ORDER BY 
+            th.transaction_date DESC;      -- Optionally order by transaction date or any other field
+    END
+    ELSE
+    BEGIN
+        -- If not an admin, return an error message
+        THROW 50000, 'Access Denied. Only admins can access transaction history.', 1;
+    END
+END;
+```
+
+### API Endpoints
+
+**Record Transaction**
+- **Endpoint:** POST /api/transactions
+- **Request Body:**  
+  ```json
+  {
+    "userID": 123,
+    "transactionGUID": "abc123",
+    "totalAmount": 100.00,
+    "transactionType": "Add",
+    "transactionDate": "2021-10-01T12:00:00",
+    "itemID": 456
+  }
+  ```
+- **Response:**  
+  ```json
+  {
+    "transactionID": 789
+  }
+  ```
+  
+**Get Transaction History**
+- **Endpoint:** GET /api/transactions?userID=123
+- **Response:**  
+  ```json
+  [
+    {
+      "transactionID": 789,
+      "userID": 123,
+      "transactionGUID": "abc123",
+      "totalAmount": 100.00,
+      "transactionType": "Add",
+      "transactionDate": "2021-10-01T12:00:00",
+      "itemID": 456
+    }
+  ]
+  ```
+
+### Data Flow
+#### When a transaction occurs (items added or checked out):
+- Front-end sends a POST request to the `/api/transactions` endpoint with transaction details and generates a TransactionGUID for the transaction.
```

Comment:
Suggestion to change this to a stored procedure so that only one api call is required to change the inventory and log a transaction at the same time.